orchagent/flexcounterorch.cpp

```diff
@@ -329,11 +330,41 @@ bool FlexCounterOrch::bake()
     return consumer->addToSync(entries);
 }
 
+static bool isCreateOnlyConfigDbBuffers(Table& deviceMetadataConfigTable)
+{
+    std::string createOnlyConfigDbBuffersValue;
+
+    try
+    {
+        if (deviceMetadataConfigTable.hget("localhost", "create_only_config_db_buffers", createOnlyConfigDbBuffersValue))
+        {
+            if (createOnlyConfigDbBuffersValue == "true")
+            {
+                return true;
+            }
+        }
+    }
+    catch(const std::system_error& e)
+    {
+        SWSS_LOG_ERROR("System error: %s", e.what());
+    }
+
+    return false;
+}
+
 map<string, FlexCounterQueueStates> FlexCounterOrch::getQueueConfigurations()
 {
     SWSS_LOG_ENTER();
 
     map<string, FlexCounterQueueStates> queuesStateVector;
+
+    if (!isCreateOnlyConfigDbBuffers(m_deviceMetadataConfigTable))
+    {
+        FlexCounterQueueStates flexCounterQueueState(0);
+        queuesStateVector.insert(make_pair(createAllAvailableBuffersStr, flexCounterQueueState));
```

Comment:
why we are not using this variable in same funtion ? why portsorch have to enable queue counter ?

Reply:
The `flexcounterorch.cpp` and the `portsorch.cpp` are related to each other.
The `flexcounterorch.cpp` is responsible for the processing of the `BUFFER_QUEUE|EthernetX` configuration and passing the `queeuStateVector` to the `portsorch.cpp` which will turn on the queue counter polling.

If we decide to turn OFF the `create_only_config_db_buffers` configuration knob, we do not need to iterate through `BUFFER_QUEUE|EthernetX` in `flexcounterorch.cpp` we need to iterate through the `m_portList` (list of all `Ethernet` ports on the switch). The `m_portList` is available only in the `portsorch.cpp`.